test/e2e_env/universal/meshaccesslog/meshaccesslog.go

```diff
@@ -120,6 +121,59 @@ spec:
 		Expect(dst).To(Equal("test-server"))
 	})
 
+	It("should log outgoing traffic with JSON formatting", func() {
+		yaml := fmt.Sprintf(`
+type: MeshAccessLog
+name: client-outgoing
+mesh: meshaccesslog
+spec:
+ targetRef:
+   kind: MeshService
+   name: demo-client
+ to:
+   - targetRef:
+       kind: Mesh
+     default:
+       backends:
+       - tcp:
+           format:
+             json:
+             - key: Source
+               value: '%%KUMA_SOURCE_SERVICE%%'
+             - key: Destination
+               value: '%%KUMA_DESTINATION_SERVICE%%'
+             - key: Start
+               value: '%%START_TIME(%%s)%%'
+           address: "%s_%s:9999"
+`, env.Cluster.Name(), externalServiceDeployment)
+		Expect(YamlUniversal(yaml)(env.Cluster)).To(Succeed())
+
+		var src, dst string
+		sinkDeployment := env.Cluster.Deployment(externalServiceDeployment).(*externalservice.UniversalDeployment)
+		Eventually(func(g Gomega) {
+			_, _, err := env.Cluster.Exec("", "", AppModeDemoClient,
+				"curl", "-v", "--fail", "test-server.mesh")
+			g.Expect(err).ToNot(HaveOccurred())
+
+			stdout, _, err := sinkDeployment.Exec("", "", "head", "-1", "/nc.out")
+			g.Expect(err).ToNot(HaveOccurred())
+
+			type log struct {
+				Source      string
+				Destination string
+				Start       string
+			}
+			var line log
+			g.Expect(json.Unmarshal([]byte(stdout), &line)).To(Succeed())
```

Comment:
Will it still work when stdout contains multiple json objects? Envoy many times just flushes many logs at once. Shouldn't we try to split by `\n` and take the first item?

Reply:
Do you mean the e2e test specifically? We call `head` to get one line here.

Reply:
my bad, didn't noticed `head`, I thought you call `cat`. Thanks!